File: rust/samples/src/layout_card_sample.rs
use fluent_cards::*;

pub fn create_two_column_card() -> Card {
    AdaptiveCardBuilder::new()
        .with_version("1.5")
        .add_text_block(|tb| {
            tb.with_text("Product Information")
                .with_size(TextSize::Large)
                .with_weight(TextWeight::Bolder);
        })
        .add_column_set(|cs| {
            cs.add_column(|col| {
                col.with_width("auto").add_image(|img| {
                    img.with_url("https://adaptivecards.io/content/adaptive-card-50.png")
                        .with_size(ImageSize::Medium);
                });
            })
            .add_column(|col| {
                col.with_width("stretch")
                    .add_text_block(|tb| {
                        tb.with_text("Adaptive Cards SDK")
                            .with_weight(TextWeight::Bolder);
                    })
                    .add_text_block(|tb| {
                        tb.with_text("Create platform-agnostic UI snippets")
                            .with_wrap(true);
                    })
                    .add_text_block(|tb| {
                        tb.with_text("$49.99")
                            .with_color(TextColor::Good)
                            .with_size(TextSize::Large);
                    });
            });
        })
        .add_action(|a| {
            a.submit("Add to Cart")
                .with_style(ActionStyle::Positive);
        })
        .build()
}

pub fn create_styled_container_card() -> Card {
    AdaptiveCardBuilder::new()
        .with_version("1.5")
        .add_container(|c| {
            c.with_style(ContainerStyle::Emphasis)
                .add_text_block(|tb| {
                    tb.with_text("Important Notice")
                        .with_size(TextSize::Large)
                        .with_weight(TextWeight::Bolder);
                })
                .add_text_block(|tb| {
                    tb.with_text(
                        "This is an emphasized section with important information.",
                    )
                    .with_wrap(true);
                });
        })
        .add_container(|c| {
            c.add_text_block(|tb| {
                tb.with_text("Regular Section")
                    .with_weight(TextWeight::Bolder);
            })
            .add_text_block(|tb| {
                tb.with_text("This is a normal section with regular styling.")
                    .with_wrap(true);
            });
        })
        .add_container(|c| {
            c.with_style(ContainerStyle::Accent)
                .add_text_block(|tb| {
                    tb.with_text("Highlighted Section")
                        .with_weight(TextWeight::Bolder);
                })
                .add_text_block(|tb| {
                    tb.with_text("This section uses accent styling to stand out.")
                        .with_wrap(true);
                });
        })
        .build()
}

pub fn create_fact_set_card() -> Card {
    AdaptiveCardBuilder::new()
        .with_version("1.5")
        .add_text_block(|tb| {
            tb.with_text("Meeting Details")
                .with_size(TextSize::Large)
                .with_weight(TextWeight::Bolder);
        })
        .add_fact_set(|fs| {
            fs.add_fact("Date", "December 15, 2024")
                .add_fact("Time", "2:00 PM - 3:00 PM")
                .add_fact("Location", "Conference Room A")
                .add_fact("Organizer", "John Smith")
                .add_fact("Attendees", "12 people");
        })
        .add_action(|a| {
            a.open_url("https://example.com/meeting/123")
                .with_title("Join Meeting");
        })
        .build()
}

pub fn create_nested_container_card() -> Card {
    AdaptiveCardBuilder::new()
        .with_version("1.5")
        .add_text_block(|tb| {
            tb.with_text("Dashboard")
                .with_size(TextSize::ExtraLarge)
                .with_weight(TextWeight::Bolder);
        })
        .add_container(|c| {
            c.with_style(ContainerStyle::Emphasis)
                .add_text_block(|tb| {
                    tb.with_text("Statistics")
                        .with_size(TextSize::Large)
                        .with_weight(TextWeight::Bolder);
                })
                .add_column_set(|cs| {
                    cs.add_column(|col| {
                        col.with_width("stretch").add_container(|cont| {
                            cont.with_style(ContainerStyle::Good)
                                .add_text_block(|tb| {
                                    tb.with_text("Active Users")
                                        .with_weight(TextWeight::Bolder);
                                })
                                .add_text_block(|tb| {
                                    tb.with_text("1,234")
                                        .with_size(TextSize::ExtraLarge);
                                });
                        });
                    })
                    .add_column(|col| {
                        col.with_width("stretch").add_container(|cont| {
                            cont.with_style(ContainerStyle::Attention)
                                .add_text_block(|tb| {
                                    tb.with_text("Pending Issues")
                                        .with_weight(TextWeight::Bolder);
                                })
                                .add_text_block(|tb| {
                                    tb.with_text("42")
                                        .with_size(TextSize::ExtraLarge);
                                });
                        });
                    });
                });
        })
        .build()
}

File: swift/Samples/layout_card_sample.swift
import FluentCards

func createTwoColumnCard() -> Card {
    return AdaptiveCardBuilder()
        .withVersion("1.5")
        .addTextBlock { tb in
            tb.withText("Product Information")
                .withSize(.large)
                .withWeight(.bolder)
        }
        .addColumnSet { cs in
            cs.addColumn { col in
                col.withWidth("auto")
                    .addImage { img in
                        img.withURL("https://adaptivecards.io/content/adaptive-card-50.png")
                            .withSize(.medium)
                    }
            }
            .addColumn { col in
                col.withWidth("stretch")
                    .addTextBlock { tb in
                        tb.withText("Adaptive Cards SDK")
                            .withWeight(.bolder)
                    }
                    .addTextBlock { tb in
                        tb.withText("Create platform-agnostic UI snippets")
                            .withWrap(true)
                    }
                    .addTextBlock { tb in
                        tb.withText("$49.99")
                            .withColor(.good)
                            .withSize(.large)
                    }
            }
        }
        .addAction { a in
            a.submit("Add to Cart").withStyle(.positive)
        }
        .build()
}

func createStyledContainerCard() -> Card {
    return AdaptiveCardBuilder()
        .withVersion("1.5")
        .addContainer { c in
            c.withStyle(.emphasis)
                .addTextBlock { tb in
                    tb.withText("Important Notice")
                        .withSize(.large)
                        .withWeight(.bolder)
                }
                .addTextBlock { tb in
                    tb.withText("This is an emphasized section with important information.")
                        .withWrap(true)
                }
        }
        .addContainer { c in
            c.addTextBlock { tb in
                tb.withText("Regular Section")
                    .withWeight(.bolder)
            }
            .addTextBlock { tb in
                tb.withText("This is a normal section with regular styling.")
                    .withWrap(true)
            }
        }
        .addContainer { c in
            c.withStyle(.accent)
                .addTextBlock { tb in
                    tb.withText("Highlighted Section")
                        .withWeight(.bolder)
                }
                .addTextBlock { tb in
                    tb.withText("This section uses accent styling to stand out.")
                        .withWrap(true)
                }
        }
        .build()
}

func createFactSetCard() -> Card {
    return AdaptiveCardBuilder()
        .withVersion("1.5")
        .addTextBlock { tb in
            tb.withText("Meeting Details")
                .withSize(.large)
                .withWeight(.bolder)
        }
        .addFactSet { fs in
            fs.addFact("Date", "December 15, 2024")
                .addFact("Time", "2:00 PM - 3:00 PM")
                .addFact("Location", "Conference Room A")
                .addFact("Organizer", "John Smith")
                .addFact("Attendees", "12 people")
        }
        .addAction { a in
            a.openURL("https://example.com/meeting/123").withTitle("Join Meeting")
        }
        .build()
}

func createNestedContainerCard() -> Card {
    return AdaptiveCardBuilder()
        .withVersion("1.5")
        .addTextBlock { tb in
            tb.withText("Dashboard")
                .withSize(.extraLarge)
                .withWeight(.bolder)
        }
        .addContainer { c in
            c.withStyle(.emphasis)
                .addTextBlock { tb in
                    tb.withText("Statistics")
                        .withSize(.large)
                        .withWeight(.bolder)
                }
                .addColumnSet { cs in
                    cs.addColumn { col in
                        col.withWidth("stretch")
                            .addContainer { cont in
                                cont.withStyle(.good)
                                    .addTextBlock { tb in
                                        tb.withText("Active Users")
                                            .withWeight(.bolder)
                                    }
                                    .addTextBlock { tb in
                                        tb.withText("1,234")
                                            .withSize(.extraLarge)
                                    }
                            }
                    }
                    .addColumn { col in
                        col.withWidth("stretch")
                            .addContainer { cont in
                                cont.withStyle(.attention)
                                    .addTextBlock { tb in
                                        tb.withText("Pending Issues")
                                            .withWeight(.bolder)
                                    }
                                    .addTextBlock { tb in
                                        tb.withText("42")
                                            .withSize(.extraLarge)
                                    }
                            }
                    }
                }
        }
        .build()
}
